woocommerce/woocommerce-ios
# [Shipping Labels] Hide contact button when phone number is missing

Closes #4659 

# Description
This PR aims to hide contact button on destination address form top banner if when customer doesn't provide phone number.

# Changes
- Added new parameter phone number to `ShippingLabelAddressTopBannerFactory`
- In the factory, hide contact button if phone number is missing.
- Updated title of banner if phone number is missing.
- Made `topBannerView` calculated variable instead of lazy to make sure that contact button shows up if phone number is entered in the form.

# Screenshots
| Phone number present | Phone number missing |
| ----- | ----- |
| <img src="https://user-images.githubusercontent.com/5533851/126464442-3f3087ae-11c0-43c1-99aa-443b95c0af82.png" width=400 /> | <img src="https://user-images.githubusercontent.com/5533851/126464414-38db537b-7ae8-4a1f-b453-148151e27219.png" width=400 /> |

# Testing
0. Make sure your test store is eligible to create shipping labels.
1. On WooCommerce web app, make phone number optional in Settings > Checkout > Billing.
2. Create an order without phone number.
3. On WCiOS app, open the new order and select Create Shipping Label.
4. On Ship To, update the address to a fake one (e.g 123 Test Street) and tap Done.
5. Notice that there's a warning banner appearing on the top without Contact Customer button.
6. Enter phone number manually in Phone field and tap Done.
7. Notice that the warning banner on the top shows Contact Customer button.

Update release notes:
